### PR TITLE
Architecture Documentation

### PR DESCRIPTION
**Pull request by achimweigel**
**What this PR does / why we need it**:

Architecture Documentation

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other developer
Architecture Documentation
```

### CODE REVIEW

File: docs/architecture/architecture-and-security.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+Image pull secrets are another group of root credentials stored in the Secret Store which must be rotated manually. 
+
+### Deploy-Pipeline
+
+This Chapter describes which credentials are created and rotated with every execution by the Deploy-Pipeline.
+
+**Credentials for GCP Project:** The Deploy-Pipeline must update the k8s secrets stored in the Gardener projects 
+Gardener-LaaS-Project and Gardener-Resource-Cluster-Project used to create and maintain shoot clusters. Therefore,
+it must store there the latest root credentials for GCP Projects stored in the Secret Store. The update of the k8s secrets
```
Comment (reshnm) [inline docs/architecture/architecture-and-security.md:182]: This sentence sounds a little bit "sperrig". Maybe you can rephrase that?

File: docs/architecture/architecture-and-security.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+- Manually rotated credentials in Secret Store
+  - Gardener-Service-Account-Kubeconfig for Gardener-LaaS-Project
+  - Gardener-Service-Account-Kubeconfig for Gardener-Resource-Cluster-Project
+  - Secret key for the GCP service account of the two GCP projects referenced by the shoot clusters
+
+- Automatically rotated by the Deploy-Pipeline
+  - Secrets containing the GCP secret keys in the Garden-Projects Gardener-LaaS-Project and 
+    Gardener-Resource-Cluster-Project referenced by shoot resources 
+  - Credentials stored in the ServiceTargetConfigs
+  - 2 Shoot-Cluster-Kubeconfigs required as Helm values to install the Central Landscaper
```
Comment (reshnm) [inline docs/architecture/architecture-and-security.md:239]: Since both of them are pointing to the same cluster, only one is needed.

**Review by reshnm — CHANGES_REQUESTED** [review]
Very good document overall 👍 
Only minor nits.

File: docs/architecture/architecture-and-security.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+- Garden-LaaS-Project: This is a Garden project where the shoot clusters with the central units of a LaaS landscape
+  are running
+
+- Core-Shoot-Cluster: A Garden shoot cluster in the Garden-LaaS-Project. On this shoot cluster the Central Landscaper 
+  and the LaaS component/instance is running. The LaaS uses the Central Landscaper to deploy the Landscaper instances for 
+  different customers.
+
+- Target-Shoot-Cluster: A Garden shoot cluster where the different Landscaper instances for the customer are running 
+  in different namespaces. In principle a LaaS landscape might have different Target-Shoot-Clusters.
```
[the remark this answers is not in the document]
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md]: I changed it

File: docs/architecture/architecture-and-security.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+Then the Deploy-Pipeline creates an Installation (and Target etc.) for the LaaS, such that the Central Landscaper
+installs it also on the Core-Shoot-Cluster.
+
+To enable the LaaS to deploy e.g. Landscaper Instances for customers to the Target-Shoot-Clusters, a 
+[ServiceTargetConfig](../usage/ServiceTargetConfigs.md) object is deployed on the Core-Shoot-Cluster for every 
+Target-Shoot-Cluster. Every ServiceTargetConfig references a secret containing a kubeconfig providing access to one 
+Target-Shoot-Cluster. The kubeconfig is created for a service account on the Target-Shoot-Cluster. 
+
+The ServiceTargetConfigs might be created manually or automatically by the Deploy-Pipeline. The only important point is 
+that the token in the kubeconfigs must be rotated with every deployment. This will be described later in more detail.
```
[the remark this answers is not in the document]
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md]: I have move it to the other chapter

File: docs/architecture/architecture-and-security.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+For every customer there is a dedicated namespace on the Core-Shoot-Cluster. To get a Landscaper instance a 
+user/customer/operator creates a [LandscaperDeployment](../usage/LandscaperDeployments.md) custom resource in this 
+namespace. The LaaS instance in the Core-Shoot-Cluster watches for such custom resources and creates an 
+[Instance](../usage/Instances.md) custom resource for every LandscaperDeployment. For every Instance, the LaaS deploys an 
+[Installation](https://github.com/gardener/landscaper/blob/master/docs/usage/Installations.md) consisting
+of a component which creates
+
+- a new shoot custom resource in the Garden-Resource-Cluster-Project for which the Gardener creates the Resource-Shoot-Cluster
+- a new namespace on one of the Target-Shoot-Clusters with a Landscaper instance.
```
[the remark this answers is not in the document]
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md]: I have changed it

File: docs/architecture/architecture-and-security.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  you have an admin Gardener-Service-Account-Kubeconfig at hand.
+
+- Shoot-Cluster-Kubeconfig: Kubeconfig with a longer lifetime and particular permissions for accessing a Gardener 
+  shoot cluster could be constructed by creating a service account (with RBAC rules) in the shoot cluster and fetching
+  kubeconfigs/token using again the 
+  [token request API](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/#TokenRequest).
+  With this approach the credentials are not stored in the shoot cluster and have a restricted lifetime. 
+  In principle, it is still possible to create the credentials by deploying a secret for a service account. Such 
+  credentials (the automatically injected token) have the disadvantages to have no restricted lifetime and exist on the 
+  client and server side. Nevertheless, they must be also be rotated (by deleting and recreating the secret).
```
[the remark this answers is not in the document]
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md:138]: Unfortunately I do not know if we could disallow tokens with unlimited lifetime. This is still possible on the virtual Garden cluster. I think token with unlimited lifetime are not a problem in general. If we remove the corresponding secret they will be invalidated. So there lifetime is still under our control. 

No I do not think we need tokens with unlimited lifetime.

A more complex problem is that if token are used by more than one application, we need to rotate them at all applications in one step or need always two of them such that we can rotate only one at one point in time. This is much easier with token having some limited lifetime.

File: docs/architecture/architecture-and-security.md (hunk 1 above)
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md]: If have tried is :-)

File: docs/architecture/architecture-and-security.md (hunk 2 above)
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md:244]: You are right. This is anyhow something which should be described in more detail. The installations of the central landscaper consists of the installations of 2 helm charts. I think this is quite important.

File: docs/architecture/architecture-and-security.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+- No static credentials for shoot clusters. (Gardener recommendation).
+- No Gardener-Service-Account with Service Account Manager role (Gardener recommendation).
+
+### Prerequisites
+
+The following summarizes the base techniques to get access to Gardener projects and shoot clusters:
+
+- Gardener-Service-Account: A project member with sufficient permissions could create Gardener-Service-Accounts with 
+  different roles for a Gardener project. For a description of the roles, see
+  [Create a Service Account](https://pages.github.tools.sap/kubernetes/gardener/docs/015-guides/gardener-api/#create-a-service-account).
```
[the remark this answers is not in the document]
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md]: I have changed this to an external link

File: docs/architecture/architecture-and-security.md (hunk 2 above)
Comment (achimweigel) [reply docs/architecture/architecture-and-security.md]: Described in more detail

**Review by robertgraeff — APPROVED** [review]

**Pull request merged**